system/alternatives.py

```diff
@@ -114,7 +124,16 @@ def main():
                         link = line.split()[1]
                         break
 
-    if current_path != path:
+    if not force:
+        need_to_force = False
+    elif not os.path.islink(link):
```

Comment:
Ah yes, you are correct, I misread this earlier.

What I was trying to get at is really the question of whether this
complexity is necessary. If "force" is passed into the module, why not just
always pass "--force" into update-alternatives?

I suspect there will be cases where you would want --force that aren't
listed here... Because we don't know about them or because the tool changes
over time.

On Thu, Jun 4, 2015, 1:00 AM James Pharaoh <notifications@github.com> wrote:

> In system/alternatives.py
> <https://github.com/ansible/ansible-modules-extras/pull/549#discussion_r31693432>
> :
>
> > @@ -114,7 +124,16 @@ def main():
> >                          link = line.split()[1]
> >                          break
> >
> > -    if current_path != path:
> > +    if not force:
> > +        need_to_force = False
> > +    elif not os.path.islink(link):
>
> No, it shouldn't do that and the first branch of the if handles that
> specifically and with the highest precedence of all the options here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ansible/ansible-modules-extras/pull/549/files#r31693432>
> .
>